### PR TITLE
WIP: Split avocado into its own test suite

### PR DESCRIPTION
**Pull request by stefwalter**
This splits avocado tests into its own test suite.

Remaining to be done: 

- [x] Blocking on #3335 ... especially where we start using new Status API contexts
- [ ] Actually run the code and make sure it works, cannot do this until above.
- [ ] Use different test runs for Firefox vs. Chrome
- [ ] Print out avocado results in their native format (if there is such a thing)

### CODE REVIEW

**Comment by dperpeet** [comment]
> Print out avocado results in their native format (if there is such a thing)

These are currently only downloaded in case of failure. You can take a look at the results directory here: https://github.com/cockpit-project/cockpit/blob/dddf51e6dd155159609a1b6310e399df9e54d788/test/avocadolib.py#L59

File: test/check-verify [hunk 1]
```diff
@@ -162,15 +84,13 @@ def main():
...
-    revision = subprocess.check_output([ "git", "rev-parse", "HEAD" ]).strip()
+    revision = os.environ.get("TEST_REVISION")
+    if not revision:
+        revision = subprocess.check_output([ "git", "rev-parse", "HEAD" ]).strip()
     name = os.environ.get("TEST_NAME")
     if not name:
         name = "test"
 
-    if opts.publish:
-        sink = start_publishing(github, opts.publish, name, revision, image)
```
Comment (mvollmer) [inline test/check-verify]: The '--publish' option should be removed as well.

File: test/check-verify [hunk 2]
```diff
@@ -19,71 +19,6 @@ EXCLUDE = [
     'check-example'
 ]
 
-def start_publishing(github, host, name, revision, image):
```
Comment (mvollmer) [inline test/check-verify]: The `github` local variable in `main` can also be removed.

**Comment by mvollmer** [comment]
I find `check-verify --publish` useful when working on support for a new OS, but that is a corner case and needs to be done carefully.  Let's worry about this the next time we need it.

File: test/github-task [hunk 3]
```diff
@@ -51,30 +166,54 @@ def main():
...
+    elif prefix == "avocado":
+        cmd = [ "./avocado/run-tests", "--install" ]
+        if opts.verbose:
+            cmd.append("--verbose")
 
     else:
         sys.stderr.write("github-task: unknown context: " + context)
+        return 1
+
+    if opts.rebase:
```
Comment (mvollmer) [inline test/github-task]: --rebase needs to be added to argparser.

**Comment by mvollmer** [comment]
Here are some fixes for github-task: https://github.com/mvollmer/cockpit/commit/4939331afaaa5b9fd4b46a927728bdcda901ca8c

**Comment by stefwalter** [comment]
Half merged, half replaced by #3505
